app/src/main/java/com/zendrive/zendrivesdkdemo/MainFragment.java

```diff
@@ -138,79 +190,59 @@ public void initializeZendriveSDK() {
 
         this.clearText();
 
-        // Get zendrive listener to catch drive start and end events
-        ZendriveListener listener = getZendriveListener();
-
         // Show loading indicator
         showLoadingIndicator();
 
-        // setup zendrive sdk
-        Zendrive.setup(
-                this.getActivity().getApplicationContext(),
-                configuration,
-                listener,
-                new Zendrive.SetupCallback() {
+        initializeZendriveSDK(getActivity().getApplicationContext(), driverId);
+    }
+
+    public void initializeZendriveSDK(Context applicationContext, final String driverId) {
+
+        ZendriveManager zendriveManager = ZendriveManager.getSharedInstance(getActivity().getApplicationContext());
+        if (zendriveManager.isSDKSetup()) {
+            return;
+        }
+        zendriveManager.initializeZendriveSDK(applicationContext, driverId, new Zendrive.SetupCallback() {
```

Comment:
remove driver id parameter, zendrivemanager should access driver id from constant class.